### PR TITLE
fix(@sap-ux/app-config-writer): dont write optional inbound config properties if not provided

### PR DESCRIPTION
**Pull request by IainSAP**
Fix for inbound-navigation config writer
- Fix for #1366 
- Update tests

### CODE REVIEW

File: .changeset/fluffy-turtles-glow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-ux/create': minor
```
Comment (IainSAP) [inline .changeset/fluffy-turtles-glow.md:2]: wrong cset

**Review by devinea — APPROVED** [review]
changeset minor ok
code changes and test updates lg.

**Review by zdravko-georgiev — APPROVED** [review]
Code changes looks good
Excellent test coverage
Didn't tested manually

**Review by slin-sap — APPROVED** [review]
- changeset looks good. It could also be considered as minor since the modified functions changed returned type definition. But patch level also seems ok since the new return type is less strict.  
- unit tests updated to cover the fixed bug scenario.
-  changes are easy to follow and reason to change has been well commented.

**Comment by IainSAP** [comment]
> * changeset looks good. It could also be considered as minor since the modified functions changed returned type definition. But patch level also seems ok since the new return type is less strict.
> * unit tests updated to cover the fixed bug scenario.
> * changes are easy to follow and reason to change has been well commented.

Thanks @slin-sap it is a minor bump :)

**Pull request merged**